File: programmers/practice-kit/brute-force/최소직사각형.py
def solution(sizes):
    width, height = [], []

    for w, h in sizes:
        width.append(max(w, h))
        height.append(min(w, h))
    
    return max(width) * max(height)



print(solution([[60, 50], [30, 70], [60, 30], [80, 40]]))

'''


'''

File: javas/최소직사각형.java
package java;

import java.util.*;

public class 최소직사각형 {
    class Solution {
        public int solution(int[][] sizes) {
            List<Integer> width = new ArrayList<Integer>();
            List<Integer> height = new ArrayList<Integer>(); 

            for(int[] size: sizes) {
                width.add(Math.max(size[0], size[1]));
                height.add(Math.min(size[0], size[1]));
            }

            int answer = Collections.max(width) * Collections.max(height);
            return answer;
        }
    }
}
